### ISSUE TITLE
Use ZFS instead of Btrfs

### ISSUE THREAD

**Issue by LukeShortCloud** [LukeShortCloud/winesapOS#543]
For the root partition.

Only make this change if ZFS on macOS and Windows becomes more stable. It is currently planned to upstream all support into the main OpenZFS project for the upcoming 3.0 release. That is slated to come out in 2023.

- https://www.phoronix.com/news/OpenZFS-3.0-Plus-Windows-Work

It will make it easier to access system files on other operating systems (mainly macOS) as only Linux and Windows support Btrfs. Using ZFS will also have the added benefit to allow users to easily add a fast L2ARC drive. We could have that configurable via the first-time setup.

Relevant projects:

- https://github.com/openzfsonosx/zfs
- https://github.com/openzfsonwindows/ZFSin
- https://github.com/maharmstone/btrfs

**Comment by LukeShortCloud** [comment]
ZFS has also been natively supported on BSD for a long time.

https://docs.freebsd.org/en/books/handbook/zfs/

Btrfs is not supported on BSD.

**Comment by LukeShortCloud** [comment]
One of these timers can be used to automatically create ZFS snapshots as a replacement to Snapper (which only supports Btrfs and LVM currently):

https://github.com/zfsonlinux/zfs-auto-snapshot

**Comment by LukeShortCloud** [comment]
As interesting of an idea that this may be, the ZFS driver will never be added natively to the Linux kernel. This means that if the DKMS module fails to build for some or all of the Linux kernels, a system may be unbootable. We have also ran into issue where DKMS modules do not build for Linux SteamOS/Neptune.

Related: https://github.com/LukeShortCloud/winesapOS/issues/532

This is not worth the risk. Closing.
